src/py/aspen/covidhub_import/implementation.py

```diff
@@ -0,0 +1,136 @@
+import warnings
+from typing import Iterable, Mapping
+
+from covid_database import init_db as Cinit_db
+from covid_database import SqlAlchemyInterface as CSqlAlchemyInterface
+from covid_database import util as Cutil
+from covid_database.models.enums import ConsensusGenomeStatus
+from covid_database.models.ngs_sample_tracking import (
+    ConsensusGenome,
+    CZBID,
+    DphCZBID,
+    Project,
+    SampleFastqs,
+)
+from sqlalchemy import and_, or_
+from sqlalchemy.orm import joinedload, Session
+
+from aspen.database.connection import session_scope, SqlAlchemyInterface
+from aspen.database.models import Group, Sample, UploadedPathogenGenome
+
+
+def covidhub_interface_from_secret(secret_id: str) -> CSqlAlchemyInterface:
+    interface = Cinit_db(Cutil.get_db_uri(secret_id))
+    return interface
+
+
+def import_project(
+    interface: SqlAlchemyInterface,
+    covidhub_secret_id: str,
+    rr_project_id: str,
+):
+    covidhub_interface = covidhub_interface_from_secret(covidhub_secret_id)
+    covidhub_session: Session = covidhub_interface.make_session()
+
+    with session_scope(interface) as session:
+        project = (
+            covidhub_session.query(Project)
+            .filter(Project.rr_project_id == rr_project_id)
+            .one()
+        )
+
+        group = (
+            session.query(Group)
+            .filter(Group.name == project.originating_lab)
+            .one_or_none()
+        )
+        if group is None:
+            # copy the project info into the "group"
+            group = Group(
+                name=project.originating_lab,
+                address=project.originating_address,
+                # FIXME: this needs to be updated.
```

Comment:
Is this going to be fixed later or did you mean to update this in the PR? If this is something we are going to update/fix later it might be good to update the note.

Reply:
https://app.clubhouse.io/genepi/story/123616/covidhub-importer-does-not-provide-a-sane-email-address-for-dph